store/store.go

```diff
@@ -1328,6 +1340,15 @@ func (s *Store) Download(ctx context.Context, name string, targetPath string, do
 
 	err = download(ctx, name, downloadInfo.Sha3_384, url, user, s, w, resume, pbar)
 	if err != nil {
+		// If sha3 checksum is incorrect and it was a resumed download, retry from scratch.
+		// Note that we will retry this way only once.
+		if _, ok := err.(HashError); ok && resume > 0 {
+			logger.Debugf("Error on resumed download: %v", err.Error())
+			err = os.Truncate(partialPath, 0)
+			if err == nil {
+				return s.Download(ctx, name, targetPath, downloadInfo, pbar, user)
```

Comment:
probably not a big issue but I find a bit confusing that on this path we will still call late the closing defer in the parent Download invocation, not sure it's easy to address though. It would be somewhat nicer to have closed w before calling Truncate